dev/TeachingTip/TeachingTip.cpp

```diff
@@ -53,62 +53,56 @@ void TeachingTip::OnApplyTemplate()
     m_tailPolygon.set(GetTemplateChildT<winrt::Polygon>(s_tailPolygonName, controlProtected));
 
-    if (auto && container = m_container.get())
+    if (auto&& container = m_container.get())
     {
         container.Child(nullptr);
     }
 
-    if (auto&& tailOcclusionGrid = m_tailOcclusionGrid.get())
+    m_contentSizeChangedRevoker = [this]()
     {
-        m_contentSizeChangedRevoker = tailOcclusionGrid.SizeChanged(winrt::auto_revoke, {
-            [this](auto const&, auto const&)
-            {
-                UpdateSizeBasedTemplateSettings();
-                // Reset the currentEffectivePlacementMode so that the tail will be updated for the new size as well.
-                m_currentEffectiveTipPlacementMode = winrt::TeachingTipPlacementMode::Auto;
-                TeachingTipTestHooks::NotifyEffectivePlacementChanged(*this);
-                if (IsOpen())
-                {
-                    PositionPopup();
-                }
-                {
-                    auto&& tailOcclusionGrid = m_tailOcclusionGrid.get();
-                    if (auto&& expandAnimation = m_expandAnimation.get())
-                    {
-                        expandAnimation.SetScalarParameter(L"Width", static_cast<float>(tailOcclusionGrid.ActualWidth()));
-                        expandAnimation.SetScalarParameter(L"Height", static_cast<float>(tailOcclusionGrid.ActualHeight()));
-                    }
-                    if (auto&& contractAnimation = m_contractAnimation.get())
-                    {
-                        contractAnimation.SetScalarParameter(L"Width", static_cast<float>(tailOcclusionGrid.ActualWidth()));
-                        contractAnimation.SetScalarParameter(L"Height", static_cast<float>(tailOcclusionGrid.ActualHeight()));
-                    }
-                }
-            }
-        });
-    }
+        if (auto && tailOcclusionGrid = m_tailOcclusionGrid.get())
+        {
+            return tailOcclusionGrid.SizeChanged(winrt::auto_revoke, { this, &TeachingTip::OnContentSizeChanged });
+        }
+        return winrt::FrameworkElement::SizeChanged_revoker{};
+    }();
 
-    if (auto && contentRootGrid = m_contentRootGrid.get())
+    if (auto&& contentRootGrid = m_contentRootGrid.get())
     {
         winrt::AutomationProperties::SetLocalizedLandmarkType(contentRootGrid, ResourceAccessor::GetLocalizedStringResource(SR_TeachingTipCustomLandmarkName));
     }
 
-    if (auto&& closeButton = m_closeButton.get())
+    m_closeButtonClickedRevoker = [this]()
     {
-        m_closeButtonClickedRevoker = closeButton.Click(winrt::auto_revoke, {this, &TeachingTip::OnCloseButtonClicked });
-    }
-    if (auto&& alternateCloseButton = m_alternateCloseButton.get())
+        if (auto&& closeButton = m_closeButton.get())
+        {
+            return closeButton.Click(winrt::auto_revoke, { this, &TeachingTip::OnCloseButtonClicked });
+        }
+        return winrt::Button::Click_revoker{};
+    }();
```

Comment:
can this just be:
```suggestion
        return {};
```

? Or does it need each return to be strongly typed because the lambda is auto return?

Reply:
I get a compiler error with the suggestion, "A brace-enclosed list does not provide a return type for this lambda"

---
In reply to: [279095260](https://github.com/Microsoft/microsoft-ui-xaml/pull/619#discussion_r279095260) [](ancestors = 279095260)